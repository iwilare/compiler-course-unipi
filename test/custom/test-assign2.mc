int main() {
    float a;

    a = 0.0;

    a += a + 10.;
    printfloat(a);

    a += a + 10.e+2;
    printfloat(a);

    a -= a + .5;
    printfloat(a);

    a *= a + 2.0;
    printfloat(a);

    a *= a + 2.2E-1;
    printfloat(a);

    a /= a + 2.;
    printfloat(a);

    return 0;
}
