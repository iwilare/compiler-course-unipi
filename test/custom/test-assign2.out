10.000000
1020.000000
-0.500000
-0.750000
0.397500
0.165798
